lib/main_common.pm

```diff
@@ -0,0 +1,219 @@
+    return 0 if get_var("DUALBOOT") || get_var("RESCUECD") || get_var("ZDUP");
+
+    return get_var("REBOOTAFTERINSTALL") && !get_var("UPGRADE");
+}
+
+sub load_login_tests {
+    if (!get_var("UEFI")) {
+        loadtest "login/boot.pm";
+    }
+}
+
+sub load_rescuecd_tests {
+    if (rescuecdstep_is_applicable()) {
+        loadtest "rescuecd/rescuecd.pm";
+    }
+}
+
+sub load_zdup_tests {
+    loadtest 'installation/setup_zdup.pm';
+    loadtest 'installation/zdup.pm';
+    loadtest 'installation/post_zdup.pm';
+    loadtest 'boot/boot_to_desktop.pm';
+}
+
+sub load_autoyast_tests {
+    #    init boot in load_boot_tests
+    loadtest("autoyast/installation.pm");
+    loadtest("autoyast/console.pm");
+    loadtest("autoyast/login.pm");
+    loadtest("autoyast/wicked.pm");
+    loadtest("autoyast/autoyast_verify.pm") if get_var("AUTOYAST_VERIFY");
+    if (get_var("SUPPORT_SERVER_GENERATOR")) {
+        loadtest("support_server/configure.pm");
+    }
+    else {
+        loadtest("autoyast/repos.pm");
+        loadtest("autoyast/clone.pm");
+        loadtest("autoyast/logs.pm");
+    }
+    loadtest("autoyast/autoyast_reboot.pm");
+    #    next boot in load_reboot_tests
+}
+
+sub installzdupstep_is_applicable {
+    return !get_var("NOINSTALL") && !get_var("RESCUECD") && get_var("ZDUP");
+}
+
+sub snapper_is_applicable {
+    my $fs = get_var("FILESYSTEM", 'btrfs');
+    return ($fs eq "btrfs" && get_var("HDDSIZEGB", 10) > 10);
+}
+
+sub gnomestep_is_applicable {
+    return check_var("DESKTOP", "gnome");
+}
+
+sub installyaststep_is_applicable {
+    return !get_var("NOINSTALL") && !get_var("RESCUECD") && !get_var("ZDUP");
+}
+
+sub bigx11step_is_applicable {
+    return get_var("BIGTEST");
+}
+
+sub noupdatestep_is_applicable {
+    return !get_var("UPGRADE");
+}
+
+sub kdestep_is_applicable {
+    return check_var("DESKTOP", "kde");
+}
+
+sub consolestep_is_applicable {
+    return !get_var("INSTALLONLY") && !get_var("DUALBOOT") && !get_var("RESCUECD");
+}
+
+sub rescuecdstep_is_applicable {
+    return get_var("RESCUECD");
+}
+
+sub remove_desktop_needles {
+    my $desktop = shift;
+    if (!check_var("DESKTOP", $desktop) && !check_var("FULL_DESKTOP", $desktop)) {
+        unregister_needle_tags("ENV-DESKTOP-$desktop");
+    }
+}
+
+our %valueranges = (
+
+    #   LVM=>[0,1],
+    NOIMAGES           => [0, 1],
+    REBOOTAFTERINSTALL => [0, 1],
+    DOCRUN             => [0, 1],
+
+    #   BTRFS=>[0,1],
+    DESKTOP => [qw(kde gnome xfce lxde minimalx textmode)],
+
+    #   ROOTFS=>[qw(ext3 xfs jfs btrfs reiserfs)],
+    VIDEOMODE => ["", "text", "ssh-x"],
+);
+
+sub check_env {
+    for my $k (keys %valueranges) {
+        next unless get_var($k);
+        unless (grep { get_var($k) eq $_ } @{$valueranges{$k}}) {
+            die sprintf("%s must be one of %s\n", $k, join(',', @{$valueranges{$k}}));
+        }
+    }
+}
+
+our @can_randomize = qw/NOIMAGES REBOOTAFTERINSTALL DESKTOP VIDEOMODE/;
+
+sub setrandomenv {
```

Comment:
updated: I couldn't possibly know if any of the parts are unused as it's triggered by an external variable and I don't know if any scenario on osd or o3 trigger this. Or *would* I be able to know?